### PR TITLE
swaps: swap long waits

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes TEAM2-455
Figma link (if any): https://www.figma.com/file/SFfYzRoHlapEecS0LTLwhQ/branch/LmnWqDhwEZq12a4lKcrtW9/Swap-Aggregator-V2?node-id=761%3A47139

## What changed (plus any additional context for devs)
we now take into account service time for cross chain swaps and display a warning if its > an hour. 

## Screen recordings / screenshots

https://cloud.skylarbarrera.com/Screen-Recording-2022-10-05-16-15-22.mp4

## What to test

regular + cross chain swaps. custom gas settings should effect gas time estimates as expected 

## Final checklist

- [ ] Assigned individual reviewers?
- [ ] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [ ] Did you test both iOS and Android?
- [ ] If your changes are visual, did you check both the light and dark themes?
- [ ] Added e2e tests? If not, please specify why
- [ ] If you added new critical path files, did you update the CODEOWNERS file?
- [ ] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: src/utils/crossChainTimeEstimates.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+export const getCrossChainTimeEstimate = ({
+  serviceTime,
+  gasTimeInSeconds,
+}: {
+  serviceTime: number;
+  gasTimeInSeconds: number;
```
Comment (estebanmino) [inline src/utils/crossChainTimeEstimates.ts:6]: nice

File: src/utils/crossChainTimeEstimates.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  const timeEstimate = serviceTime + gasTimeInSeconds;
+
+  const minutes = Math.floor(timeEstimate / 60);
+  const hours = Math.floor(minutes / 60);
+
+  if (hours >= 1) {
+    isLongWait = true;
+    timeEstimateDisplay = `>${hours} ${hours === 1 ? 'hr' : 'hrs'} 􀇿`;
+  } else if (minutes >= 1) {
+    timeEstimateDisplay = `~${minutes} min}`;
```
Comment (estebanmino) [inline src/utils/crossChainTimeEstimates.ts:23]:
```suggestion
    timeEstimateDisplay = `~${minutes} min`;
```

**Review by estebanmino — CHANGES_REQUESTED** [review]
https://www.loom.com/share/033e7acfd0714bb088c4b789709119e1

UI looking nice on both android and ios 🚀 

i did find something weird when displaying secs tho, you can see on the video that i'm seeing 1215 secs as time estimate, feels off

**Comment by skylarbarrera** [comment]
> https://www.loom.com/share/033e7acfd0714bb088c4b789709119e1
> 
> UI looking nice on both android and ios 🚀
> 
> i did find something weird when displaying secs tho, you can see on the video that i'm seeing 1215 secs as time estimate, feels off

looks like this is cause you're overriding one value but not the source. without your change this would get parsed and  displayed as min or hrs

**Comment by estebanmino** [comment]
oh ok let me try again

**Comment by estebanmino** [comment]
is constantly crashing now https://www.loom.com/share/fb86965665f1487284989199d705befd

**Comment by skylarbarrera** [comment]
current explainer while we wait for updated copy: https://cloud.skylarbarrera.com/Screen-Recording-2022-10-11-11-41-48.mp4

**Comment by skylarbarrera** [comment]
updated PoW: https://cloud.skylarbarrera.com/Screen-Shot-2022-10-14-12-14-08.44.png

File: src/screens/ExchangeModal.tsx [hunk 3]
```diff
@@ -1280,6 +1280,9 @@ export default function ExchangeModal({
                 marginBottom={0}
                 marginTop={0}
                 testID={`${testID}-gas`}
+                crossChainServiceTime={
+                  (tradeDetails as CrosschainQuote)?.routes?.[0]?.serviceTime
```
Comment (estebanmino) [inline src/screens/ExchangeModal.tsx:1284]: do you want to create a helper to get this param maybe like https://github.com/rainbow-me/rainbow/blob/develop/src/handlers/swap.ts#L101 

i see this being used in different places

File: src/components/gas/GasSpeedButton.js [hunk 4]
```diff
@@ -107,13 +109,16 @@ const GasSpeedPagerCentered = styled(Centered).attrs(() => ({
...
   const { colors } = useTheme();
   return (
     <Label
       align="right"
+      // eslint-disable-next-line no-nested-ternary
       color={
-        theme === 'dark'
+        isLongWait
+          ? colors.lightOrange
+          : theme === 'dark'
```
Comment (estebanmino) [inline src/components/gas/GasSpeedButton.js:121]: 😅  wdyt of moving `color` definition above the return so we don't ignore the warning?

**Comment by estebanmino** [comment]
two minor comments but looking great on both android and ios https://www.loom.com/share/4f05b657a60e40e6803fc5d11e9cf355 light and dark mode

File: src/screens/ExchangeModal.tsx (hunk 3 above)
Comment (skylarbarrera) [reply src/screens/ExchangeModal.tsx:1284]: yeah agree

**Review by estebanmino — APPROVED** [review]
🚀 🚀 🚀 🚀 🚀

**Review by derHowie — APPROVED** [review]
lgtm
